arch/arm/include/aarch32/cortex_m/tz_ns.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief
+ */
+
+#ifndef ZEPHYR_ARCH_ARM_INCLUDE_AARCH32_CORTEX_M_TZ_NS_H_
+#define ZEPHYR_ARCH_ARM_INCLUDE_AARCH32_CORTEX_M_TZ_NS_H_
+
+#ifdef _ASMLANGUAGE
+
+/* nothing */
+
+#else
+
+/**
+ * @brief Macro for "sandwiching" a function call (@p name) in two other calls
+ *
+ * This macro should be called via @ref __TZ_WRAP_FUNC.
+ *
+ * This macro creates the function body of an "outer" function which behaves
+ * exactly like the wrapped function (@p name), except that the preface function
+ * is called before, and the postface function afterwards.
+ *
+ * @param preface   The function to call first. Must have no parameters and no
+ *                  return value.
+ * @param name      The main function, i.e. the function to wrap. This function
+ *                  will receive the arguments, and its return value will be
+ *                  returned.
+ * @param postface  The function to call last. Must have no parameters and no
+ *                  return value.
+ * @param store_lr  The assembly instruction for storing away the LR value
+ *                  before the functions are called. This instruction must leave
+ *                  r0-r3 unmodified.
+ * @param load_lr   The assembly instruction for restoring the LR value after
+ *                  the functions have been called. This instruction must leave
+ *                  r0-r3 unmodified.
+ */
+#define __TZ_WRAP_FUNC_RAW(preface, name, postface, store_lr, load_lr) \
+	do { \
+		__asm(".global "#preface"; .type "#preface", %function"); \
+		__asm(".global "#name"; .type "#name", %function"); \
+		__asm(".global "#postface"; .type "#postface", %function"); \
+		\
+		__asm(store_lr); \
+		__asm("push {r0-r3}"); \
+		__asm("bl "#preface); \
+		__asm("pop {r0-r3}"); \
+		__asm("bl "#name); \
+		__asm("push {r0-r3}"); \
+		__asm("bl "#postface); \
+		__asm("pop {r0-r3}"); \
+		__asm(load_lr); \
+	} while (0)
+
+/**
+ * @brief Macro for "sandwiching" a function call (@p name) in two other calls
+ *
+ * This is for functions which do not pass arguments or return values via the
+ * stack. I.e. the arguments and return values must each fit within 4 words,
+ * after accounting for alignment.
+ *
+ * If nothing is passed on the stack, the stack can be used to store LR without
+ * messing with offsets.
+ *
+ *	int foo(char *arg); // Implemented elsewhere.
```

Comment:
nit: can you replace the // with /* */? Here and in the lines below

Reply:
It's already inside a /**/, so that won't work.